buildtools/buckaudit/buckaudit.go

```diff
@@ -0,0 +1,149 @@
+package buckaudit
+
+import (
+	"encoding/json"
+
+	"github.com/apex/log"
+
+	"github.com/fossas/fossa-cli/api/fossa"
+	"github.com/fossas/fossa-cli/errors"
+	"github.com/fossas/fossa-cli/exec"
+	"github.com/fossas/fossa-cli/graph"
+	"github.com/fossas/fossa-cli/pkg"
+)
+
+type buckOutput struct {
+	outputMapping map[string][]string
+}
+
+// Deps returns the dependencies of a Buck project using the buck audit command
+func Deps(name string) (graph.Deps, error) {
+	// Upload the dependencies.
+	locatorMap, err := uploadDeps(name)
```

Comment:
It probably makes more sense to refactor these steps into:

1. Make all the `buck` calls you need, and then build a struct describing the dependency graph and its files.
2. Traverse the dependency graph, doing the uploading and FOSSA-graph-building/import-setting stuff for each dependency in one go.

This way you don't need to carry around a `locatorMap`, which seems error-prone.